tools/oecert/host/host.cpp

```diff
@@ -250,18 +251,20 @@ static char* _get_enclave_filename()
         goto done;
     }
     path[path_size] = '\0';
-
-    enclave_filename =
-        (char*)malloc(path_size + sizeof(ENCLAVE_FILENAME_SUFFIX));
+    enclave_filename_size = path_size + sizeof(ENCLAVE_FILENAME_SUFFIX);
+    enclave_filename = (char*)malloc(enclave_filename_size);
 
     if (enclave_filename == nullptr)
     {
         printf("Failed to malloc enclave_filename.\n");
         goto done;
     }
-
-    strcpy(enclave_filename, path);
-    strcat(enclave_filename, ENCLAVE_FILENAME_SUFFIX);
+    snprintf(
```

Comment:
This is still not safe, since `snprintf()` does not guarantee null termination.  
See https://en.cppreference.com/w/c/io/fprintf and https://stackoverflow.com/questions/1270387/are-snprintf-and-friends-safe-to-use
`snprintf_s()` on the other hand does safe truncation.

Reply:
👍

Reply:
I checked again and `snprintf()` guarantees null termination. Microsoft's `_snprintf()` does not though.
https://stackoverflow.com/questions/7706936/is-snprintf-always-null-terminating
I also tested it on both linux and windows by truncating the buffer size, and the null terminator was included at both platforms.